src/Data.Json.JSemantic.purs

```diff
@@ -0,0 +1,105 @@
+module Data.Json.JSemantic
+  ( JSemantic(..)
+  , toSemantic
+  ) where
+
+import Data.Argonaut.JCursor(JsonPrim(..), runJsonPrim)
+import Data.Maybe
+import Data.String.Regex (RegexFlags(..), regex, test)
+import Data.String
+import Data.Foldable
+import qualified Data.Array as A
+import qualified Data.Array.Unsafe as AU
+import Data.Tuple
+import qualified Data.Date as Date
+import Math
+
+
+data JSemantic = Integral   Number
+               | Fractional Number
+               | Date       Date.Date
+               | DateTime   Date.Date
+               | Time       Date.Date
+               | Interval   Date.Date Date.Date
```

Comment:
That might be the case for durations, but an Interval (as suggested in the spec) is "part of the time axes limited by two instants" ([wp](http://en.wikipedia.org/wiki/Time#Time-like_concepts:_terminology)), and two dates are an adequate representation.

We should choose a `(date|time|datetime)Regex` that is in line with ISO8601 but is sufficiently strict (eg. won't parse not fully specified dates like simple numbers). 

We can also make regexes configurable, and split JSemantic into a separate library. Are you ok with that?